pkg/option/config.go

```diff
@@ -3570,6 +3590,8 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	// To support K8s NetworkPolicy
 	c.EnableK8sNetworkPolicy = vp.GetBool(EnableK8sNetworkPolicy)
 	c.PolicyCIDRMatchMode = vp.GetStringSlice(PolicyCIDRMatchMode)
+	c.EnablePerNodeIdentity = vp.GetBool(EnablePerNodeIdentity)
```

Comment:
`EnablePerNodeIdentity` depends on `PolicyCIDRMatchMode` having "nodes" set as a value to work. You correctly make this check in another place, but we should log a warning here if `EnablePerNodeIdentity` is set to "true" but `PerNodeIdentitiesEnabled` returns "false".

Reply:
Hmm I don't think it depends on `PolicyCIDRMatchMode` in any way. It only uses the same ID scope. About the `PerNodeIdentitiesEnabled` are actually returning `c.EnablePerNodeIdentity` -- so are the same thing. 

```
func (c *DaemonConfig) PerNodeIdentitiesEnabled() bool {
	return c.EnablePerNodeIdentity
}
```

Reply:
> Hmm I don't think it depends on PolicyCIDRMatchMode in any way. It only uses the same ID scope. About the PerNodeIdentitiesEnabled are actually returning c.EnablePerNodeIdentity -- so are the same thing.

I made a correction to my original comment.

`PolicyCIDRMatchesNodes` is the function that enabled per-node identity. This PR (and the flag `EnablePerNodeIdentity` contained in it) do not actually enable per-node identity.  Those node identities will not be present unless  the config function `PolicyCIDRMatchesNodes` returns "true". Otherwise all internal nodes get an identity of "2".